distribution/Dockerfile

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM centos:7
+
+# Project version defined in pom.xml is passed as an argument
+ARG VERSION
+
+# JDK 8 is a pre-requisite to run Drill ; 'which' package is needed for drill-config.sh
+RUN yum install -y java-1.8.0-openjdk-devel which ; yum clean all ; rm -rf /var/cache/yum
+
+# The drill tarball is generated upon building the Drill project
+COPY target/apache-drill-$VERSION.tar.gz /tmp
+
+# Drill binaries are extracted into the '/opt/drill' directory
+RUN mkdir /opt/drill
+RUN tar -xvzf /tmp/apache-drill-$VERSION.tar.gz --directory=/opt/drill --strip-components 1
+
+# Starts Drill in embedded mode and connects to Sqlline
+ENTRYPOINT /opt/drill/bin/drill-embedded
```

Comment:
The `Dockerfile` does not expose ports:

```
EXPOSE 8047/tcp
```

The above is for the web console port. Should we expose others?

Reply:
For embedded mode, i've not needed to expose the ports. For K8S, i've needed to expose the HTTP port (8047) and the server port (31010) to enable the loadbalancer.

Reply:
Even in embedded mode, I'd definitely want to see the query profile, so no harm in exposing the web port.

Yes, for production, we need to expose the three Drill ports. (As an aside, this means the ports can't be customized. Which is fine: if someone wants to run multiple Drillbits per node, they need to use an overlay network such as Weave, Calico, etc.)

Reply:
I meant to say that i've not needed to expose the ports to have the web UI or any functionality working. I've tried on Mac and Linux. I've only needed to expose the ports in K8S env when using an externally accessible IP via LoadBalancer. 

And yes, I use weave for K8S. It's helpful for multi-node Docker env on K8S.